src/bridge-canister/src/bridge.rs

```diff
@@ -0,0 +1,90 @@
+#![allow(async_fn_in_trait)]
+
+use bridge_did::error::{BftResult, Error};
+use bridge_did::op_id::OperationId;
+use bridge_did::order::SignedMintOrder;
+use bridge_utils::bft_events::{self, BurntEventData, MintedEventData, NotifyMinterEventData};
+use bridge_utils::evm_bridge::EvmParams;
+use bridge_utils::evm_link::EvmLink;
+use candid::CandidType;
+use did::{H160, H256};
+use eth_signer::sign_strategy::TransactionSigner;
+use ic_task_scheduler::task::TaskOptions;
+use serde::{Deserialize, Serialize};
+
+pub trait Operation:
+    Sized + CandidType + Serialize + for<'de> Deserialize<'de> + Clone + Send + Sync + 'static
+{
+    async fn progress(self, id: OperationId, ctx: impl OperationContext) -> BftResult<Self>;
+    fn is_complete(&self) -> bool;
+
+    /// Address of EVM wallet to/from which operation will move tokens.
+    fn evm_address(&self) -> H160;
+
+    fn scheduling_options(&self) -> Option<TaskOptions> {
+        Some(TaskOptions::default())
+    }
+
+    async fn on_wrapped_token_minted(
+        _ctx: impl OperationContext,
+        _event: MintedEventData,
+    ) -> Option<OperationAction<Self>> {
+        None
```

Comment:
I removed the default implementation.

But I think it is better to keep the standard `Option`, because it is much easier to use: a lot of utility methods, question mark operator, etc...